### PR TITLE
 modified emc_ice2ioda.py to output ioda_v2 and created test files

### PR DESCRIPTION
**Pull request by YanHao-NOAA**
## Description

Updated:
Variable dimensions dictionary
Dimensions dictionary
Global attributes dictionary
Remove record numbers
Update keyDict
Units, Fill values, other metadata
Update CMakeLists.txt with environment variable
Added test reference file for emc_ice.

### Issue(s) addressed:  #655 

Link the issues to be closed with this PR
- fixes issue #655

## Dependencies

- [ ] merge before #686

### CODE REVIEW

File: src/marine/emc_ice2ioda.py [hunk 1]
```diff
@@ -118,16 +116,22 @@ def main():
...
 
     # write them out
-    AttrData['date_time_string'] = fdate.strftime("%Y-%m-%dT%H:%M:%SZ")
+    GlobalAttrs['date_time_string'] = fdate.strftime("%Y-%m-%dT%H:%M:%SZ")
+    ObsVars, nlocs = iconv.ExtractObsData(ice.data, locationKeyList)
+    DimDict = {'nlocs': nlocs}
+    writer = iconv.IodaWriter(args.output, locationKeyList, DimDict)
+
+    VarAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
+    VarAttrs[('sea_ice_area_fraction', 'ObsValue')]['units'] = 'unitless'
```
[the remark this answers is not in the document]
Comment (YanHao-NOAA) [reply src/marine/emc_ice2ioda.py]: Done, thanks!

File: src/marine/emc_ice2ioda.py [hunk 2]
```diff
@@ -118,16 +116,22 @@ def main():
...
+    VarDims = {
+        'sea_ice_area_fraction': ['nlocs'],
+    }
     # Read in
-    ice = Observation(args.input, args.thin, fdate, writer)
+    ice = Observation(args.input, args.thin, fdate)
 
     # write them out
-    AttrData['date_time_string'] = fdate.strftime("%Y-%m-%dT%H:%M:%SZ")
+    GlobalAttrs['date_time_string'] = fdate.strftime("%Y-%m-%dT%H:%M:%SZ")
```
[the remark this answers is not in the document]
Comment (YanHao-NOAA) [reply src/marine/emc_ice2ioda.py]: Removed date_time_string.

**Review by rhoneyager — APPROVED** [review]
Looks good. Thanks for adding the test!

**Comment by YanHao-NOAA** [comment]
I will modify the cryosat_ice after this one is merged to develop.

**Comment by YanHao-NOAA** [comment]
> Looks good. Thanks for adding the test!

Ryan, thank you for your review!

**Review by srherbener — APPROVED** [review]

**Pull request merged**